docs/src/proposals/interest-bearing-tokens.md

```diff
@@ -0,0 +1,284 @@
+---
+title: Interest-Bearing Tokens
+---
+
+This document describes a solution for interest-bearing tokens as an on-chain
+program. Interest-bearing tokens continuously accrue interest as time passes.
```

Comment:
i-tokens are also extensible for tokens with lockup and more customized tokens?

Reply:
You would probably create another token program for those special cases.  This i-token proposal is really two things:

* the i-token program, which is just for interest-bearing tokens
* updating the overall programming model to suit N token programs

So to answer your question, I wouldn't expand the i-tokens for lockup / more customization, but once the programming model is updated, it'll be possible to create another token program for things like lockup.